src/tests.rs

```diff
@@ -232,15 +232,23 @@ fn panic_in_child() {
         ()
     })
     .unwrap_err();
-    eprintln!("error: {error:?}",);
-    assert!(matches!(
-        error,
-        MemIsolateError::CallableStatusUnknown(
-            CallableStatusUnknownError::CallableProcessDiedDuringExecution
-        )
-    ));
+    eprintln!("error: {error:?}");
+    if cfg!(target_os = "macos") {
+        assert!(matches!(
+            error,
+            MemIsolateError::CallableStatusUnknown(
+                CallableStatusUnknownError::ChildProcessKilledBySignal(5),
```

Comment:
What causes MacOS to do this? If we remove the `c::mock::disable_mocking()` block included in this PR, we actually see it in a few more tests as well.